basic_crud/basic_crud/basic_crud.py

```diff
@@ -0,0 +1,202 @@
+    "code":"",
+    "label":"",
+    "image":"/favicon.ico",
+    "quantity":0,
+    "category":"",
+    "seller":"",
+    "sender":""
+}"""
+
+URL_OPTIONS = {
+    "GET": "products",
+    "POST": "products",
+    "PUT": "products/{pr_id}",
+    "DELETE": "products/{pr_id}",
+}
+
+
+class State(rx.State):
+    """The app state."""
+
+    products: list[Product]
+    _db_updated: bool = False
+
+    def load_product(self):
+        with rx.session() as session:
+            self.products = session.query(Product).all()
+        yield State.reload_product
+
+    @rx.background
+    async def reload_product(self):
+        while True:
+            await asyncio.sleep(2)
+            if self.db_updated:
+                async with self:
+                    with rx.session() as session:
+                        self.products = session.query(Product).all()
+                    self._db_updated = False
+
+    @rx.var
+    def db_updated(self):
+        return self._db_updated
+
+    @rx.var
+    def total(self):
+        return len(self.products)
+
+
+class QueryState(State):
+    body: str = DEFAULT_BODY
+    response_code: str = ""
+    response: str = ""
+    method: str = "GET"
+    url_query: str = URL_OPTIONS["GET"]
+    query_options = list(URL_OPTIONS.keys())
+
+    def update_method(self, value):
+        if self.url_query == "":
+            self.url_query = URL_OPTIONS[value]
+        self.method = value
+
+    @rx.var
+    def need_body(self):
+        return False
+
+    @rx.var
+    def f_response(self):
+        return f"""```json\n{self.response}\n```"""
+
+    def clear_query(self):
+        self.url_query = URL_OPTIONS["GET"]
+        self.method = "GET"
+        self.body = DEFAULT_BODY
+
+    async def send_query(self):
+        url = f"http://localhost:8000/{self.url_query}"
+        async with httpx.AsyncClient() as client:
+            match self.method:
+                case "GET":
+                    res = await client.get(url)
+                case "POST":
+                    res = await client.post(url, data=self.body)
+                case "PUT":
+                    res = await client.put(url, data=self.body)
+                case "DELETE":
+                    res = await client.delete(url)
+                case _:
+                    res = None
+        self.response_code = res.status_code
+        if self.response_code == 200:
+            self.response = json.dumps(res.json(), indent=2)
+            self._db_updated = True
+        else:
+            self.response = res.content.decode()
+
+
+def data_display():
+    return rx.vstack(
+        rx.heading(State.total, " products found"),
+        rx.foreach(State.products, render_product),
+        rx.spacer(),
+        width="30vw",
+        height="100%",
+    )
+
+
+def render_product(product: Product):
+    return rx.hstack(
+        rx.image(src=product.image, height="100%", width="3vw"),
+        rx.text("(", product.code, ") ", product.label, width="10vw"),
```

Comment:
```suggestion
        rx.text(f"({product.code}) {product.label}", width="10vw"),
```

i personally prefer the f-string in situations like this

Reply:
Yeah I took a bad habit back when f-string were not supported in this place.